### DISCUSSION TITLE
Utilize All Nodes in a Canvas Group (and Nested Groups) as Context

### DISCUSSION THREAD

**Discussion by VentureTactics** [rpggio/obsidian-chat-stream#25] [Ideas]
## Summary
When connecting a group as an ancestor to a User Prompt node, apply all nodes and files within the group to the context, ordered by the X,Y coordinates (up to down, and left to right).

## Reason
There are many instances where it'd be much easier to add nodes to a group, then connect the group to a Chat Stream conversation, than to try and connect each individual node.

## Useful Features
- This should apply to all groups nested within the connected group as well

## Illustration
See an example of the current state in the below image:
![Testing Chat Stream -- Group Embeds](https://github.com/rpggio/obsidian-chat-stream/assets/88736327/325614ee-4032-4418-9300-00b0a209278a)

**Comment by rpggio** [comment]
Yes, the feature makes sense. Intuitively, I think this could solve some context-management features that folks have had trouble with when using arrow connections only.

**Reply by VentureTactics** [reply]
It also could build on top of the canvas node merging idea.
